lib/session.ex

```diff
@@ -0,0 +1,195 @@
+defmodule Janus.Session do
+  @moduledoc """
+  Keeps information about session and sends keep-alive messages
+  if given connection's transport requires it.
+  """
+
+  @type t :: pid()
+  @type message_t :: map()
+  @type connection_t :: pid()
+  @type timeout_t :: non_neg_integer()
+  @type session_id_t :: non_neg_integer()
+  @type plugin_t :: String.t()
+
+  @type plugin_handle_id :: pos_integer
+
+  @default_timeout 5000
+
+  use GenServer
+  alias Janus.Connection
+
+  @doc """
+  Synchronously creates a new session on the gateway.
+
+  ## Arguments
+
+  * `connection` - a PID of the `Janus.Connection` process,
+  * `timeout` - a timeout for the call.
+
+  ## Return values
+
+  On success it returns `{:ok, session}` where session is
+  a pid of `Janus.Session` process which keeps track of
+  session indentifier used by the gateway.
+
+  On error it returns `{:error, reason}`.
+
+  The reason might be:
+
+  * `{:gateway, code, info}` - it means that the call itself succeded but the
+    gateway returned an error of the given code and info,
+  * other - some serious error happened.
+  """
+  @spec create_linked_session(connection_t(), timeout_t()) :: {:ok, Janus.Session.t()} | {:error, any}
```

Comment:
Catch positive error. If Janus handles the response ok, but it cant succeed it sends something like:
```elixir
{:ok,  %{"error" => _message, "error_code" => @no_such_feed_error}}
```

Reply:
Do we have anywhere listed `Janus` error codes?

Reply:
In Janus codebase, good luck reading C source code. (look for a lot of numeric constants) Maybe we should move this into a separate PR with a standardized error handler?

Reply:
#4